ntoskrnl/io/iomgr/iofunc.c

```diff
@@ -2192,7 +2193,8 @@ NtQueryInformationFile(IN HANDLE FileHandle,
     else
     {
         /* Validate the information class */
-        if ((FileInformationClass >= FileMaximumInformation) ||
+        if ((FileInformationClass < 0) ||
```

Comment:
From the N/X/D-DK definitions:
```c
typedef enum _FILE_INFORMATION_CLASS {
  FileDirectoryInformation = 1,
...
  FileMaximumInformation
} FILE_INFORMATION_CLASS, *PFILE_INFORMATION_CLASS;
}
```
So, even checking against zero is not correct for this case.
Maybe using something like `MIN_FILE_INFO_CLASS` as done in [here](https://git.reactos.org/?p=reactos.git;a=blob;f=ntoskrnl/ex/sysinfo.c;hb=c09a7b5bd70a8a3926b434316e431a9acd5dab15#l2820) for example could be done instead of hardcoding the name of the first enum value?

Same comment applies for all the FileInformationClass pieces of code you have modified here.

Reply:
The fact that `IopQueryOperationLength[0] == 0` will take care of that.